File: DatasetCreation/src/scr/Host.java
package scr;

import java.net.InetAddress;

public class Host 
{
    /*
     * Questa classe rappresenta un host identificato da un indirizzo IP e una porta.
     * Fornisce metodi per ottenere e impostare l'indirizzo e la porta dell'host.
     */

    // Indirizzo IP dell'host
    private InetAddress addr;
    // Porta dell'host
    private int port;

    public Host() throws Exception
    {
        this.addr = InetAddress.getByName("0.0.0.0");
        this.port = 0;
    }

    public Host(InetAddress addr, int port)
    {
        this.addr = addr;
        this.port = port;
    }

    public InetAddress getAddress()
    {
        return this.addr;
    }

    public void setAddress(InetAddress addr)
    {
        this.addr = addr;
    }

    public int getPort()
    {
        return this.port;
    }

    public void setPort(int port)
    {
        this.port = port;
    }

}

File: ModelImplemetantion/Python/MLP/UDP/Host.py
# La classe Host rappresenta un host di rete con un indirizzo e una porta.

import socket

class Host:
    def __init__(self, addr: str, port: int):
        self.addr = addr
        self.port = port

    def get_address(self) -> str:
        return self.addr

    def set_address(self, addr: str):
        self.addr = addr

    def get_port(self) -> int:
        return self.port

    def set_port(self, port: int):
        self.port = port